### DISCUSSION TITLE
NPMPlus w/o network_mode: host

### DISCUSSION THREAD

**Discussion by FlattusBlastus** [nextcloud/all-in-one#3558] [Ideas] [answered]
@Zoey2936 
Submitting a thought for your review:
Due to need to reserve certain ports, network_mode: host doesnt seem to offer any advantages over:
    ports:
      - 80:80
      - 81:81 #Port for the Admin Panel
      - 443:443

Everything else can be set / modified in the .env
Removing the network_mode: host would allow the container to be used in Windows Docker Desktop.

EDIT: I don't mean to come across so rude sometimes. I find myself testing NC AIO in the evenings after dealing with security issues all day. :-)

**Comment by Zoey2936** [comment, marked as answer]
doing so would cause the following:
- no IP binding options
- can't proxy to other docker containers, without adding NPMplus to their network
- IPv6 in docker needs correct setup (should be done even with network mode host)
- crowdsec also needs to run in docker and NPMplus inside that network
- the reverse proxy can't connect to other hosts outside docker bound to localhost
- etc.

There could maybe two versions of the community container one network mod ehost and one without, but not sure if that would makes sense

**Reply by FlattusBlastus** [reply]
Not sure about the IP binding options but my usual NPM uses the default bridge network.
It can proxy to other docker containers without being in their network.
Doesn't need IPv6.
Not sure about crowdsec, I'd have to read up.
And connecting to hosts _in front_ of the proxy is bad practice. Only connect to hosts behind the proxy.

version: "3.8"
services:
  nginx_proxy_manager:
    image: jc21/nginx-proxy-manager:latest
    restart: unless-stopped
    container_name: nginx-proxy-manager
    ports:
      - 80:80
      - 81:81 #Port for the Admin Panel
      - 443:443
    environment:
      - DISABLE_IPV6=true
    volumes:
      - /c/dockerdata/npm:/data
      - /c/dockerdata/npm/le:/etc/letsencrypt
    labels:
      - "com.centurylinklabs.watchtower.enable=true"

**Reply by Zoey2936** [reply]
1. NPM doesn't support IP-Bindings in network_mode host, NPMplus does
2. maybe this is only possible on windows
3. I would always enable IPv6
4. there would be then the need of a community container with crowdsec in it...
5. There are still aplication that don't have a docker image and need to run on localhost
